nsis/src/nsis/NsisAnalyzer.java

```diff
@@ -145,4 +148,44 @@ private void resolveStrings(Instruction instr, MemoryBlock stringsBlock)
 		}
 	}
 
+	/**
+	 * Resolve the control flow for the specified instruction
+	 * 
+	 * @param instr        the instruction
+	 * @param entriesBlock the memory block containing the instructions
+	 * @param program
+	 * @throws MemoryAccessException
+	 */
+	private void resolveControlFlow(Instruction instr, MemoryBlock entriesBlock,
+			ReferenceManager referenceManager) throws MemoryAccessException {
+		String mnemonic = instr.getMnemonicString();
+		int instructionNumber;
+		switch (mnemonic) {
+		case "Call":
```

Comment:
I'm not sure the best way to organize a switch statement like this. We could potentially need a clause for every instruction.

Using an enum is an option: https://stackoverflow.com/a/6392518

Probably best to get the switch logic right when we only have 3 so it's not a huge refactor later.

Reply:
Good idea. I also like the design in nrs, where a string is declared for each instruction and indicates the format of the arguments. Ex 'SSJI' would mean arg1:string arg2:string arg3:jump and arg4:int https://github.com/isra17/nrs/blob/master/nrs/ida/proc.py#538

Reply:
Would such an enum be used only in this switch or in other places as well?

Reply:
It would be used in other places, such as when resolving string parameters for each instruction

Reply:
Good solution: use inheritance to deal with resolving parameters for each instruction (like I did to deal with the compression). I'll refactor the strings resolving and the control flow after this is merged